library/Fission/Platform/Heroku/Provision/Request/Types.hs

```diff
@@ -0,0 +1,70 @@
+module Fission.Platform.Heroku.Provision.Request.Types (Request (..)) where
+
+import           Data.UUID as UUID
+import           Data.Swagger hiding (name)
+
+import           Fission.Prelude
+import qualified Fission.Plan.Types                   as Plan
+import qualified Fission.Platform.Heroku.Region.Types as Heroku
+
+data Request = Request
+  { callbackUrl :: Text          -- ^ The URL which should be used to retrieve updated information about the add-on and the app which owns it.
+  , name        :: Text          -- ^ Logical name of the resource being provisioned.
+  -- , oauthGrant :: Maybe Text -- OAuthGrant -- ^ OAuth object details (nullable).
```

Comment:
Whys this commented out?

Reply:
Well, we're not doing anything with the grant, it's _technically_ optional. We do want to start using it in the near term, but the blocker has been this PR. It's a whole other data structure, which isn't really that much work to add. The commented out key is mainly a reminder that it lives there. I'll open an Issue for that 👍

Reply:
For reference, here's an example provision request payload from the Heroku docs:

```
{
  "options": {},
  "oauth_grant": {
    "code": "01234567-89ab-cdef-0123-456789abcdef",
    "type": "authorization_code",
    "expires_at": "2016-03-03T18:01:31-0800"
  },
  "plan": "basic",
  "region": "amazon-web-services::us-east-1",
  "uuid": "01234567-89ab-cdef-0123-456789abcdef"
}
```